src/components/ModalTextEditor/ModalTextEditor.jsx

```diff
@@ -0,0 +1,158 @@
+import "./modalTextEditor.css";
+
+const ModalTextEditor = () => {
+  const [paleteVisible, setPaleteVisible] = useState(false);
+  const [priorityVisible, setPriorityVisible] = useState(false);
+  const {
+    authUser: { token },
+  } = useAuth();
+
+  const {
+    initalInput,
+    setInitialInput,
+    dispatchNote,
+    isNoteEditing,
+    initalFormValues,
+    setIsNoteEditing,
+    settextEditorVisible,
+    noteState: { notes },
+  } = useNotes();
+
+  // add to note
+
+  const handleAddToNote = (e) => {
+    e.preventDefault();
+    initalInput["createdTime"] = new Date().toLocaleString();
+    addNotes(initalInput, token, dispatchNote, (status) => {
+      if (status === 201 || 200) {
+        settextEditorVisible(false);
+        setInitialInput(initalFormValues);
+        setIsNoteEditing(false);
+      }
+    });
+  };
+
+  // edit note api
+  const handleEditNote = (e) => {
+    e.preventDefault();
+    if (isNoteEditing) {
+      initalInput["createdTime"] = new Date().toLocaleString();
+      editNotes(token, initalInput, dispatchNote, (status) => {
+        if (status === 201 || 200) {
+          settextEditorVisible(false);
+          setInitialInput(initalFormValues);
+          setIsNoteEditing(false);
+          toast.success("Note Update Successfully");
+        }
+      });
+    }
+  };
+
+  const isNoteExits = isArrayExits(notes, initalInput._id);
+  console.log("isNoteExits", isNoteExits);
+
+  return (
+    <>
+      <div className="editor-body">
+        <div
+          className="modal-editor"
+          style={{
+            backgroundColor: initalInput.bgColor,
+            color: initalInput.bgColor === "#16504b" ? "white" : "black",
+          }}
+        >
+          <form>
+            <div className="bd-bottom">
+              <input
+                value={initalInput.noteTitle}
+                onChange={(e) =>
+                  setInitialInput({ ...initalInput, noteTitle: e.target.value })
+                }
+                type="text"
+                className="big-input"
+                placeholder="Enter note title"
+              />
+            </div>
+            <TextEditor
+              value={initalInput.content}
+              setValue={(e) => setInitialInput({ ...initalInput, content: e })}
+            />
+            <div className="bd-top">
+              <input
+                type="text"
+                className="big-input"
+                placeholder="Add your lable"
+                value={initalInput.lable}
+                onChange={(e) =>
+                  setInitialInput({ ...initalInput, lable: e.target.value })
+                }
+              />
+            </div>
+            <div className="editor-footer">
+              <div className="editor-option">
+                <i
+                  role={"button"}
+                  className="fa-solid fa-palette"
+                  onClick={() => {
+                    console.log("open note palate color");
```

Comment:
you can remove the console log from here as well since you are already setting the visibility of the palette using state.